src/ray/protobuf/monitor.proto

```diff
@@ -22,9 +22,22 @@ message GetRayVersionReply {
   string version = 1;
 }
 
+message DrainAndKillNodeRequest {
+  // The node ids to drain.
+  repeated bytes node_ids = 1;
+}
+
+message DrainAndKillNodeReply {
+  // The node ids which are beginning to drain.
+  repeated bytes drained_nodes = 2;
+}
+
 // This service provides a stable interface for a monitor/autoscaler process to interact
 // with Ray.
 service MonitorGcsService {
   // Get the ray version of the service.
   rpc GetRayVersion(GetRayVersionRequest) returns (GetRayVersionReply);
+  // Request that GCS drain and kill a node. This call is idempotent, and could
+  // need to be retried if the head node fails.
+  rpc DrainAndKillNode(DrainAndKillNodeRequest) returns (DrainAndKillNodeReply);
```

Comment:
The comment on `gcs_service.proto` says 

```


// Drain a node from GCS Service.
--
  |   | // When the RPC is replied,
  |   | // - It is guaranteed that the requested nodes are going to be drained eventually.
  |   | // - It is guaranteed that GCS has persisted the draining states.
  |   | // - It is **not** guarateed that nodes receive the drain requests from GCS.


```

But I don't think that's accurate (especially in GCS FT world because the rpc seems to return before the KV ops are committed?

Reply:
If this is true, that's a bug. It can only return after commit

Reply:
hmmm, do you mean the method is supposed to be an async request and is supposed to return asap even before the function finished?

Reply:
After reading the code, I think it'll be better if we return after storing the data in the table. Without that, it guarantees nothing even the request returned successfully

Reply:
I guess when I wrote this I didn't assume GCS FT. I think returning after storing the data to GCS makes more sense.